docs/src/pages/components/skeleton/SkeletonChildren.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import Box from '@material-ui/core/Box';
+import Typography from '@material-ui/core/Typography';
+import Avatar from '@material-ui/core/Avatar';
+import Grid from '@material-ui/core/Grid';
+import { makeStyles } from '@material-ui/core/styles';
+import Skeleton from '@material-ui/lab/Skeleton';
+
+const useStyles = makeStyles(() => ({
+  image: {
+    width: '100%',
+  },
+}));
+
+function SkeletonChildrenDemo(props: { loading?: boolean }) {
+  const { loading = false } = props;
+  const classes = useStyles();
+
+  return (
+    <div>
+      <Box display="flex" alignItems="center">
+        <Box margin={1}>
+          {loading ? (
+            <Skeleton variant="circle">
+              <Avatar />
+            </Skeleton>
+          ) : (
+            <Avatar src="https://pbs.twimg.com/profile_images/877631054525472768/Xp5FAPD5_reasonably_small.jpg" />
+          )}
+        </Box>
+        <Box width="100%">
+          {loading ? (
+            <Skeleton width="100%">
+              <Typography>.</Typography>
+            </Skeleton>
+          ) : (
+            <Typography>Ted</Typography>
+          )}
+        </Box>
+      </Box>
+      {loading ? (
+        <Skeleton variant="rect" width="100%">
+          <img
+            className={classes.image}
+            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAJCAQAAACRI2S5AAAAEklEQVR42mNk+M+AFzCOKgADALyGCQGyq8YeAAAAAElFTkSuQmCC"
```

Comment:
On second thought: I don't think this is a "good" use case. This requires a placeholder image that immediately gets dimensions. Otherwise you'll get a layout shift. At that point you might as well specify the dimensions explicitly which is way more descriptive than using this blob.

It makes more sense to use the actual image here so that the browser can warm-up the request or even lazy load the image. But I don't think this is what this example is trying to demonstrate. It looks like the skeleton is displayed while the image is loaded in the background which is solvable with explicit `width` and `height` props.

Reply:
Much agree

Reply:
Agreed. I tried making it clear that I'm not sure whether it's right or wrong in the original issue, by saying that I'm new to this component / skeletons in general.

But for this, it was more about keeping the aspect ratio, which would be better served by [the `aspect-ratio` property](https://developer.mozilla.org/en-US/docs/Web/CSS/aspect-ratio).

> It makes more sense to use the actual image here so that the browser can warm-up the request or even lazy load the image. But I don't think this is what this example is trying to demonstrate.

Exactly yeah, it's not what this was trying to demonstrate. Like in an app, you may not know what "the actual image" even is to do any warm-up.